aws-crt-kotlin/native/src/aws/sdk/kotlin/crt/io/EventLoopGroupNative.kt

```diff
@@ -16,12 +19,42 @@ import aws.sdk.kotlin.crt.Closeable
  * Otherwise, maxThreads will be the number of event loops in the group.
  * @throws [aws.sdk.kotlin.crt.CrtRuntimeException] If the system is unable to allocate space for a native event loop group
  */
-public actual class EventLoopGroup actual constructor(maxThreads: Int) : Closeable, AsyncShutdown {
+@OptIn(ExperimentalForeignApi::class)
+public actual class EventLoopGroup actual constructor(maxThreads: Int) : CrtResource<aws_event_loop_group>(), Closeable, AsyncShutdown {
+    private val elg: CPointer<aws_event_loop_group>
+
+    override val ptr: CPointer<aws_event_loop_group>
+        get() = elg
+
+    private val shutdownCompleteChannel = Channel<Unit>(Channel.RENDEZVOUS)
+    private val channelStableRef = StableRef.create(shutdownCompleteChannel)
```

Comment:
**Question**: I don't understand what this channel is used for. It's a rendezvous channel so either `send` or `receive` suspends until the other is called, right? I see a `send` in `onShutdownComplete` but no `receive` call anywhere...won't this just suspend indefinitely?

Reply:
I commented elsewhere, it should be waited on in `waitForShutdown`. 

It won't block in `onShutdownComplete` because it's using `trySend`.